include/oxen/quic/btstream.hpp

```diff
@@ -280,5 +285,7 @@ namespace oxen::quic
         size_t parse_length(std::string_view req);
 
         size_t num_pending_impl() const { return user_buffers.size(); }
+
+        size_t num_awaiting_response_impl() const { return sent_reqs.size(); }
```

Comment:
perhaps this should be sent + unsent, in the off chance that one calls this just as a request is queued to be sent.

Reply:
That's a good idea, but i feel it makes more sense to add to `::num_pending()`? I can see adding it to `::num_awaiting_response()` defeating the purpose of differentiating them in the first place

If we hadd your suggestion to `::num_pending()`, the application could just go
`bs->num_awaiting_response() - bs->num_pending()` since `Stream::unsent()` is just `user_buffers.size() - unacked_size`?

Reply:
So I guess there's three values of interest here: number of sent pending requests, number of unsent pending requests, and their sum.  I think all three could arguably be useful for different reasons.

Reply:
I can see that, but isn't the third case doable with the first two?
* num_pending = user_buffers.size();
  - this is the number (not total size) of unsent requests (this is also verified in test case 004) 
* num_awaiting_response = sent_reqs.size();
  - this is the number of successfully sent requests pending response or timeout
* The sum of the above two would be identical to summing the results from the invocations, like `bs->num_awaiting_response() + bs->num_pending()`

There are already two IOChannel methods that inherit through for the values in your first comment (the latter is called `::unsent()` as well, but returns the total data size)

Reply:
Yes, *but* by doing two separate calls as the user you make them not atomic so the sum can change between calls.  That said, I don't think that's really a problem in practice; it should be fine as it is.